### PR TITLE
YACHT-1054: Introducing filtering for tables which are modified till …

### PR DESCRIPTION
**Pull request by radkomateusz**
…last census snapshot. Saving Backup Quality SLI history.

### CODE REVIEW

File: src/slo/backup_quality/quality_sli_service.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+        logging.info("Quality SLI tables filtered from %s to %s", len(all_tables), len(filtered_tables))
+        self.streamer.stream(
+            filtered_tables,
+            snapshot_marker=self.__create_snapshot_marker_row(snapshot_time)
+        )
+
+    def __should_stay_as_sli_violation(self, table):
+        try:
+            return not self.table_newer_modification_predicate.is_modified_till_last_census_snapshot()
```
Comment (jarekdrabek) [inline src/slo/backup_quality/quality_sli_service.py]: I think it should be 'is_modified_since_last_census_snapshot'

File: src/slo/sli_handlers.py [hunk 2]
```diff
@@ -14,13 +14,19 @@ def get(self):
...
+        ]
+        tasks.append(
             Tasks.create(
                 method='POST',
-                url='/sli/latency_for_x_days',
-                params={'x_days': x_days})
-            for x_days in [0, 3, 4, 5, 7]
-        ]
+                url='/sli/quality',
+                params={}))
```
Comment (jarekdrabek) [inline src/slo/sli_handlers.py]: I believe, you don't need to provide params if its empty

File: src/slo/sli_view_querier.py [hunk 3]
```diff
@@ -16,11 +16,3 @@ def query(self):
         logging.info("Executing query: %s", query)
         query_results = self.big_query.execute_query(query)
         return self.query_specification.format_query_results(query_results, self.snapshot_time), self.snapshot_time
-
-    @staticmethod
-    def sli_entry_to_table_reference(table):
```
Comment (jarekdrabek) [inline src/slo/sli_view_querier.py]: I'm wondering why you moved it from here to both specification duplicating the same code? What was the reason?

Comment (radkomateusz) [reply src/slo/sli_view_querier.py]: answered in comment under Przemek's comment

File: tests/slo/backup_quality/test_latency_query.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
+import unittest
```
Comment (jarekdrabek) [inline tests/slo/backup_quality/test_latency_query.py]: I'm wondering if the name of the file is correct :D

Comment (radkomateusz) [reply tests/slo/backup_quality/test_latency_query.py]: done

File: tests/slo/backup_quality/test_quality_sli_service.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
+import unittest
```
Comment (jarekdrabek) [inline tests/slo/backup_quality/test_quality_sli_service.py]: I would change all 'till' to 'since'

File: src/slo/sli_handlers.py (hunk 2 above)
Comment (radkomateusz) [reply src/slo/sli_handlers.py]: done

File: tests/slo/backup_quality/test_quality_sli_service.py (hunk 5 above)
Comment (radkomateusz) [reply tests/slo/backup_quality/test_quality_sli_service.py]: done

File: src/slo/backup_quality/quality_sli_service.py (hunk 1 above)
Comment (radkomateusz) [reply src/slo/backup_quality/quality_sli_service.py]: done

File: src/slo/backup_creation_latency/latency_query_specification.py [hunk 6]
```diff
@@ -23,3 +24,10 @@ def format_query_results(self, results, snapshot_time):
                               "backupLastModified": float(result['f'][7]['v']),
                               "xDays": self.x_days} for result in results]
         return formatted_results
+
+    @staticmethod
+    def latency_sli_entry_to_table_reference(table):
```
[the remark this answers is not in the document]
Comment (przemyslaw-jasinski) [reply src/slo/backup_creation_latency/latency_query_specification.py]: Alright, I have only 1 more point, how about renaming method to 'entry_to_table_reference' ?

**Pull request merged**